src/librustc_middle/ty/mod.rs

```diff
@@ -1046,6 +1046,51 @@ impl<'tcx> Predicate<'tcx> {
     pub fn kind(self) -> &'tcx PredicateKind<'tcx> {
         &self.inner.kind
     }
+
+    /// Returns the inner `PredicateAtom`.
```

Comment:
We should probably expand on this to say that this atom may contain various bound variables that reference to binders introduced by this predicate or inner predicates.